tools/small_rna_map/small_rna_map.py

```diff
@@ -0,0 +1,82 @@
+import os
+from os.path import basename
+import pysam
+import argparse
+import numpy
+import collections
+
+
+def Parser():
+        the_parser = argparse.ArgumentParser()
+        the_parser.add_argument('--input', action="store", type=str, help="input BAM file")
+        the_parser.add_argument('--output', action="store", type=str, help="output tabular file")
+
+        args = the_parser.parse_args()
+        return args
+
+
+def Ref_length(bamfile):
+	
+	""" 
+	bamfile.references: list of reference names, and bamfile.lengths is list of their lengths 
+	"""
+        ref_length={}
+        for e1, e2 in zip(bamfile.references, bamfile.lengths):
+		ref_length[e1]=e2
+        return ref_length
+
+
+def Ref_coord_pol(bamfile):
+        ref_coord ={}
+
+        for read in bamfile:
+                if not read.is_unmapped:
+			polarity = "F" if (not read.is_reverse) else "R"
+                        read_rname = read.reference_name
+                        read_position = read.pos + 1
+                        the_key = (read_rname, read_position,polarity)
+                        try:
+                                ref_coord[the_key] = ref_coord[the_key] + 1
+                                
+                        except:
+                                ref_coord[the_key] = 1
+        return collections.OrderedDict(sorted(ref_coord.items())) #dictionary {(read_rname,read_position,polarity):nmbr_read}
```

Comment:
My bad, you're right actually !
Yet the structure of the dictionary should be {(read_rname,read_position,polarity):[readsize1, readsize2, ...]}
had the end, the length of the list gives you the number of reads, and you can easily compute statistics on the list (median, mean of the read lengths at a given position)